trappy/bare_trace.py

```diff
@@ -134,5 +134,9 @@ def add_parsed_event(self, name, dfr, pivot=None):
 
     def finalize_objects(self):
         for trace_class in self.trace_classes:
+            trace_class.tracer = self
             trace_class.create_dataframe()
             trace_class.finalize_object()
+
+    def generate_data_dict(self):
+        return None
```

Comment:
If we remove it, it will break code that calls it (since not all tracers are expected to implement it). Also not implementing it is not an exception but the normal fall back.